cpp/CMakeLists.txt

```diff
@@ -262,9 +262,12 @@ if(ARROW_BUILD_TESTS OR ARROW_BUILD_BENCHMARKS)
   set(ARROW_WITH_ZLIB ON)
 endif()
 
-if(ARROW_BUILD_TESTS)
-  # JSON parsing of arrays is required for Arrow unit tests
-  set(ARROW_IPC ON)
+if(ARROW_BUILD_TESTS AND NOT ARROW_JSON)
```

Comment:
`ARROW_BUILD_INTEGRATION` also requires JSON (since the integration testing data files are in the JSON format).